src/com/gh4a/activities/SearchActivity.java

```diff
@@ -375,4 +454,35 @@ public void onItemClick(Object item) {
             startActivity(IntentUtils.getUserActivityIntent(this, user.getLogin(), user.getName()));
         }
     }
+
+
+    private class SaveSearchSuggestionTask extends BackgroundTask<Void> {
+
+        public SaveSearchSuggestionTask() {
+            super(SearchActivity.this);
+        }
+
+        @Override
+        protected Void run() throws IOException {
+            saveSuggestion(mSearchType.getSelectedItemPosition(), mQuery);
```

Comment:
Type and suggestion should be passed into the task constructor instead of getting them from member variables to avoid threading issues.